node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -19,63 +19,20 @@ use syn::Ident;
 
 use super::*;
 
-/// Returns all combinations for a single replacement:
-/// 1. generic args with `NEW` in place
-/// 2. subsystem type to be replaced
-/// 3. the subsystem name to be replaced by a new type and value
-/// 4. all other subsystems that are supposed to be kept
-fn derive_replacable_generic_lists(
-	info: &OverseerInfo,
-) -> Vec<(TokenStream, Ident, Ident, Vec<Ident>)> {
-	// subsystem generic types
-	let builder_generic_ty = info.builder_generic_types();
-
-	let to_be_replaced_name = info.subsystem_names_without_wip();
-	let baggage_generic_ty = &info.baggage_generic_types();
-
-	builder_generic_ty
-		.iter()
-		.enumerate()
-		.map(|(idx, to_be_replaced_ty)| {
-			let mut to_keep_name = to_be_replaced_name.clone();
-			let to_be_replaced_name: Ident = to_keep_name.remove(idx);
-
-			let mut builder_generic_ty = builder_generic_ty.clone();
-			builder_generic_ty[idx] = format_ident!("NEW");
-
-			let generics_ts = quote! {
-				<S, #( #baggage_generic_ty, )* #( #builder_generic_ty, )* >
-			};
-
-			(generics_ts, to_be_replaced_ty.clone(), to_be_replaced_name, to_keep_name)
-		})
-		.collect::<Vec<(_, _, _, _)>>()
-}
-
 /// Implement a builder pattern for the `Overseer`-type,
 /// which acts as the gateway to constructing the overseer.
 ///
 /// Elements tagged with `wip` are not covered here.
 pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 	let overseer_name = info.overseer_name.clone();
-	let builder = Ident::new(&(overseer_name.to_string() + "Builder"), overseer_name.span());
-	let handle = Ident::new(&(overseer_name.to_string() + "Handle"), overseer_name.span());
-	let connector = Ident::new(&(overseer_name.to_string() + "Connector"), overseer_name.span());
+	let builder = format_ident!("{}Builder", overseer_name);
+	let handle = format_ident!("{}Handle", overseer_name);
+	let connector = format_ident!("{}Connector", overseer_name);
 
 	let subsystem_name = &info.subsystem_names_without_wip();
-	let subsystem_name_init_with = &info
-		.subsystem_names_without_wip()
-		.iter()
-		.map(|subsystem_name| format_ident!("{}_with", subsystem_name))
-		.collect::<Vec<_>>();
-	let subsystem_name_replace_with = &info
-		.subsystem_names_without_wip()
-		.iter()
-		.map(|subsystem_name| format_ident!("replace_{}", subsystem_name))
-		.collect::<Vec<_>>();
-
-	let builder_generic_ty = &info.builder_generic_types();
+	let subsystem_type = &info.subsystem_generic_types();
 
+	let consumes = &info.consumes_without_wip();
```

Comment:
The `wip` denotes that the subsystem is stilll work in progress, and hess no channels will be hooked up. So the name conveys a bit of a skewed meaning.